controller/src/main/java/org/jboss/as/controller/operations/global/ReadResourceHandler.java

```diff
@@ -506,6 +508,8 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                 ModelNode value = ar.response;
                 if (!value.has(FAILURE_DESCRIPTION)) {
                     sortedAttributes.put(entry.getKey(), value.get(RESULT));
+                    if(value.hasDefined(RESPONSE_HEADERS))
+                    context.addResponseWarning(Level.WARNING, value.get(RESPONSE_HEADERS));
```

Comment:
Please indent and wrap the body of the if in curly braces.

I think propagation needs to be more complex, i.e. IIRC the headers from the value have structure that includes the level and a message so that info needs to be used to create equivalent output here.